sandy/core/cov.py

```diff
@@ -514,29 +544,93 @@ def get_covariance(cls, std):
         Example
         -------
         >>> S = pd.Series(np.array([0, 2, 3]), index=pd.Index([1, 2, 3]))
-        >>> cov = sandy.CategoryCov.get_covariance(S)
+        >>> cov = sandy.CategoryCov.from_var(S)
         >>> cov
                     1           2           3
         1 0.00000e+00 0.00000e+00 0.00000e+00
         2 0.00000e+00 2.00000e+00 0.00000e+00
         3 0.00000e+00 0.00000e+00 3.00000e+00
 
         >>> assert type(cov) is sandy.CategoryCov
+
+        >>> S = sandy.CategoryCov.from_var((1, 2, 3))
+        >>> S
+                    0           1           2
+        0 1.00000e+00 0.00000e+00 0.00000e+00
+        1 0.00000e+00 2.00000e+00 0.00000e+00
+        2 0.00000e+00 0.00000e+00 3.00000e+00
+
+        >>> assert type(S) is sandy.CategoryCov
+        >>> assert type(sandy.CategoryCov.from_var([1, 2, 3])) is sandy.CategoryCov
+        """
+        if hasattr(var, "__len__") and type(var) != str:
+            if type(var) == pd.Series:
+                cov = pd.DataFrame(np.diag(var.values),

```

Comment:
are you sure this doesn't work with `np.diag(var)`?

Reply:
Works

Reply:
It cant be done: with np.diag, the previous index are lost